docs/parameters.md

```diff
@@ -0,0 +1,324 @@
+# Passing parameters to brokers
+
+Table of Contents
+- [Overview](#overview)
+- [Design](#design)
+  - [Basic example](#basic-example)
+  - [Passing sensitive data](#passing-sensitive-data)
+  - [Value types](#value-types)
+    - [String](#string)
+    - [JSON object](#json-object)
+    - [Other primitive types (int, bool, float)](#other-primitive-types-int-bool-float)
+    - [Arrays](#arrays)
+  - [Passing all parameters from a single source at once](#passing-all-parameters-from-a-single-source-at-once)
+  - [Conflict resolution](#conflict-resolution)
```

Comment:
> proposed API changes

I wanted this to be a final design document, where API changes are irrelevant (this is a temporary information). 

> how we'll detect parameter changes for instances

currently we don't detect any changes, and I'm not sure if it makes sense to tell the user to rename his secrets if he wants to propagate the changes. I mean, it's up to the user to find an appropriate workaround, and I don't want us to be responsible for the guidance there.